### DISCUSSION TITLE
No records matched your filter criteria. (data table filter result)

### DISCUSSION THREAD

**Discussion by ishah-twilio** [twilio-labs/paste#943] [General] [answered]
Hello,

Any guidance on if we should / can use warning alert to show "No records matched your filter criteria." for data table query?

![Screen Shot 2020-11-16 at 11 28 41 AM](https://user-images.githubusercontent.com/69016180/99298660-e9a4cc00-27fe-11eb-91ca-b6c45a6d439a.png)

**Comment by kate-porter** [comment, marked as answer]
Hey @ishah-twilio - the Alert component should generally be reserved for system-level notifications, as specified in the [docs](https://paste.twilio.design/components/alert). Since this is not a system-level notification, I wouldn't recommend using the Alert.

We don't have a pattern for this defined yet, so I would encourage you to see how this is being handled in other parts of the product. I suspect a composition using Card would work for you, as this is effectively an empty state.

**Reply by ishah-twilio** [reply]
Thank you. I will research other products and will report back as appropriate.

**Comment by ishah-twilio** [comment]
Thank you @katieporter .

Looked at messaging insights examples -- and updated based on your recommendation. 
Is it ok to use primary button action to clear filter in card component? 
Also Filter section has button link to clear filters - should it be icon or button?

<img width="858" alt="Screen Shot 2020-11-16 at 2 25 07 PM" src="https://user-images.githubusercontent.com/69016180/99315481-8ffccb80-2817-11eb-9a1f-0edc66695b86.png">

**Reply by kate-porter** [reply]
Nice, this treatment looks good.

If there is no other primary button on the page, then it's fine to use a primary button as the "clear" action in the card component. That probably is the primary action for the user at this point, since that's the only way to unblock them, so that makes sense to me.

For the "clear" action in the filters section - I'd say that's fine as-is, I don't think an icon would be as clear. Keep in mind that using the link-style button does come with accessibility tradeoffs (see the last bullet point [here](https://paste.twilio.design/components/button#accessibility)).
